clients/src/test/java/org/apache/kafka/clients/MockClient.java

```diff
@@ -117,11 +119,22 @@ public void blackout(Node node, long duration) {
         blackedOut.put(node, time.milliseconds() + duration);
     }
 
+    public void authenticationFailed(Node node, long duration) {
+        authenticationException.put(node, (AuthenticationException) Errors.SASL_AUTHENTICATION_FAILED.exception());
+        disconnect(node.idString());
+        blackout(node, duration);
+    }
+
+    public void authenticationSucceeded(Node node) {
```

Comment:
Doesn't seem any point to this function if we don't return anything. Maybe we can just remove it?

Reply:
This method was removed in the commit, so the behavior matches `NetworkClient`'s; i.e. exceptions won't clear after the blackout period.